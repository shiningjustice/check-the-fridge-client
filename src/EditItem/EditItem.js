import React, { Component } from 'react';

import ApiContext from '../ApiContext';

export default class EditItem extends Component {
	static contextType = ApiContext;

  state = {
		error: null,
		name: '', 
		sectionId: '',
		quantity: '',
		note: '',
  };
  
  render() {
    return(
      <div className={`EditItem__div ${this.props.className}`}>

      </div>
    )
  }
}