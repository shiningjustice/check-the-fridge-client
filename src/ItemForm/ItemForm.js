import React, { Component } from 'react';
// import PropTypes from 'prop-types';

import config from '../config';
import ApiContext from '../ApiContext';

import './ItemForm.css'

const Required = () => (
  <span className='ItemForm__Required'>*</span>
)

export default class ItemForm extends Component {  

  static contextType = ApiContext;

  state = {
		error: null,
		name: '', 
		sectionId: '',
		quantity: '',
		note: '',
  };
  
  handleInputChange = e => {
		const target = e.target;
		const value = target.type === 'checkbox' ? target.checked : target.value;
		const name = target.name;

		console.log(name, value)
		this.setState({
			[name]: value
		});
  }
  
	resetState = () => {
		this.setState({
			error: null,
			name: '', 
			sectionId: '',
			quantity: '',
      note: '',
      missingFields: [],
		})
	}

  handleClickCancel = () => {
		this.resetState();
		this.props.history.push('/dashboard');
  };

  addMissingField = field => {
    this.setState({
      missingFields: [...this.state.missingFields, field]
    });
  }

  validateOnSubmit = () => {
		const requiredFields = ['name', 'sectionId', 'quantity'];

    for (let i = 0; i > requiredFields.length; i++) {
      if (!requiredFields[i]) {
        this.addMissingField(requiredFields[i])
      }
    }
		// requiredFields.map(field => {
		// 	if (!this.state[field].value) {
		// 		this.addMissingField(field)
		// 	}
		// });
  };
  
  handleAddSubmit = e => {
		e.preventDefault();

		this.validateOnSubmit();

		const item = {
			name: this.state.name,
			sectionId: this.state.sectionId,
			quantity: this.state.quantity,
			note: this.state.note
		};
		console.log(item)
		this.setState({ error: null });

		fetch(`${config.API_ENDPOINT}/items`, {
			method: "POST",
			body: JSON.stringify(item),
			headers: {
				"content-type": "application/json"
			},
		})
			.then(res => {
				if (!res.ok) {
					return res.json().then(e => Promise.reject(e));
				}
				return res.json();
      })
			.then(data => {
				this.resetState();
				this.props.formName === 'addItem' ? this.context.addItem(data) : this.context.editItem(data)
				this.props.history.push('/dashboard');
			})
			.catch(error => {
				console.error(error);
				this.setState({ error });
			});
  };
  
  render() {
    const { error } = this.state;
    const { sections = [] } = this.context;
    const { formName, formVerb } = this.props;

		const { item } = this.props;
		
		console.log(item)
    // const name = item.name;
    // const sectionId = item.sectionId;
    // const quantity = item.quantity;
    // const note = item.note;

		console.log(item)
		
    return (
			<section className={formName}>
				<h2>{formVerb} Item</h2>
				<form 
					className={`${formName}__form`}
          onSubmit={(formName === 'Add')
            ? e => this.handleAddSubmit(e)
            : e => this.handleEditSubmit(e)
          }
				>
					<div className={`${formName}__error`} role='alert'>
            {error && <p>{error.message}</p>}
            {this.state.missingFields &&
							`Please fill out the following required fields and resubmit: ${this.state.missingFields}`}
					</div>
					

					<div>
						<label htmlFor='name'>
              Name
              <Required />
            </label>
						{' '}
						<input
							type='text'
							name='name'
              id='name'
              // defaultValue={name && name}
							value={this.state.name.value}
							onChange={e => this.handleInputChange(e)}
							required
						></input>
					</div>

					<div>
						<label htmlFor='sectionId'>
              Section
              <Required />  
            </label>
						{' '}
						<select
							name='sectionId'
							id='sectionId'
							value={this.state.sectionId.value}
							onChange={e => this.handleInputChange(e)}
							required
						>
							<option value=''>Select one</option>
							{sections.map(section => (
								<option key={section.id} value={section.id}>
									{section.name}
								</option>
							))}
						</select>
					</div>
					
					<div>
						<label htmlFor='quantity'>
              Quantity
              <Required />  
            </label>
						{' '}
						<input
							type='number'
							name='quantity'
							id='quantity'
							value={this.state.quantity.value}
							onChange={e => this.handleInputChange(e)}
							required
						></input>
					</div>

					<div>
						<label htmlFor='quantity'>Note</label>
						{' '}
						<textarea
							type='text'
							name='note'
							id='note'
							value={this.state.note.value}
							onChange={e => this.handleInputChange(e)}
						></textarea>
					</div>

					<div className={`${formName}__buttonContainer`}>
            <button onClick={this.handleClickCancel}>Cancel</button>
            {' '}
            <button type='submit'>{`Save`}</button>
					</div>
				</form>
			</section>
		);
  }
}
 