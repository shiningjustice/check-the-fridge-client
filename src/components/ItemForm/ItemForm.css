.AddItem__section.mainContainer,.EditItem__section.mainContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 3rem;
}

.AddItem__form, .EditItem__form {
  clear: both;
  width: 200px;
  text-align: left;
}

.AddItem__form > *, .EditItem__form > * {
  margin-bottom: .5rem;
}

.ItemForm__Required{
  color: #a51414;
}

label {
  margin-bottom: .3rem;
}

.AddItem__input,.EditItem__input {
  width: 100%;
  clear: both;
  margin-bottom: .5rem;
}

.AddItem__textarea, .EditItem__textarea, .AddItem__select, .EditItem__select {
  border-radius: 5px;
  width: 100%;
  clear: both;
  height: 1.5rem;
}

.AddItem__buttonContainer, .EditItem__buttonContainer {
  display: flex;
  justify-content: space-evenly;
}

@media screen and (min-width: 768px) {
  .AddItem__section.mainContainer,.EditItem__section.mainContainer {
    /* margin-left: auto;
    margin-right: auto; */
    padding: 30px 20px;
  }
}
