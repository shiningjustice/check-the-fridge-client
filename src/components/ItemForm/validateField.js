import React from 'react';

export default function ValidationError(props) {
  if (props) {
    return (
      <div className='error'>{`Item ${props} is required`}</div>
    )
  }
  
  return;
}