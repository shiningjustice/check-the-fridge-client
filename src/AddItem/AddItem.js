import React from "react";

import ItemForm from '../ItemForm/ItemForm'

export default function AddItem (props) {
	console.log(props.history)
	return (
		<ItemForm
			formName='AddItem'
			formVerb='Add'
			history={props.history}
		/>
	)
}