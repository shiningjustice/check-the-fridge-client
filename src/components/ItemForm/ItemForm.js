/*******************************************************************
	IMPORTS
*******************************************************************/
//Libraries
import React, { Component } from 'react';
import moment from 'moment';
// Contexts and services
import ApiContext from '../../contexts/ApiContext';
import ItemsApiService from '../../services/items-api-service';
// CSS
import './ItemForm.css'

// Other Components
const Required = () => (
  <span className='ItemForm__Required'>*</span>
)

/*******************************************************************
	`ItemForm` COMPONENT
*******************************************************************/
export default class ItemForm extends Component {  
	// Context
  static contextType = ApiContext;

  state = {
		error: null,
		id: '',
		name: '', 
		sectionId: '',
		dateAdded: moment().format('YYYY-MM-DD'),
		quantity: '',
		note: '',
  };
  
	resetState = () => {
		this.setState({
			error: null,
			name: '', 
			sectionId: '',
			dateAdded: moment().format('YYYY-MM-DD'),
			quantity: '',
      note: '',
      missingFields: [],
		})
	}

  addMissingField = field => {
    this.setState({
      missingFields: [...this.state.missingFields, field]
    });
  }

  validateOnSubmit = () => {
		const requiredFields = ['name', 'dateAdded','sectionId', 'quantity'];

    for (let i = 0; i > requiredFields.length; i++) {
      if (!requiredFields[i]) {
        this.addMissingField(requiredFields[i])
      }
    }
		// requiredFields.map(field => {
		// 	if (!this.state[field].value) {
		// 		this.addMissingField(field)
		// 	}
		// });
	};
	
	/*******************************************************************
    HANDLER FUNCTIONS
  *******************************************************************/
	handleInputChange = e => {
		let value = e.target.value;
		let name = e.target.name;
		
		this.setState({
			[name]: value
		});
	}
	
	handleClickCancel = e => {
		e.preventDefault();
		this.resetState();
		this.props.history.push('/demo');
  };
  
	
  handleSubmit = e => {
		e.preventDefault();

		this.validateOnSubmit();

		const { name, sectionId, dateAdded, quantity, note, id } = this.state;
		let item = {
			name,
			sectionId: +sectionId,
			quantity,
			dateAdded: moment(dateAdded).format(),
			note,
			id
		};

		this.setState({ error: null });

		//Choose fetch call based on form
		(this.props.formName === 'AddItem') ? this.handlePost(item) : this.handlePatch(item);
	};

	handlePost = item => {
		ItemsApiService.postItem(item)
			.then(data => {
				this.resetState();
				this.context.addItem(data)
				this.props.history.push('/demo');
			})
			.catch(error => this.setState({ error: error.error}))
	}

	handlePatch = item => {
		ItemsApiService.patchItem(item, this.state.id)
			.then(() => {
				this.resetState();
				this.context.editItem(item)
				this.props.history.push('/demo');
			})
			.catch(error => this.setState({ error: error.error}))
	}

	/*******************************************************************
    LIFECYCLE METHODS
  *******************************************************************/
	componentDidMount = () => {
		// Set up form to submit on enter keypress
		window.addEventListener("keypress", this.handlePressEnter);

		const { item } = this.props;

		if (item) {
			this.setState({
				id: item.id,
				name: item.name, 
				sectionId: (item.sectionId).toString(),
				dateAdded: moment(item.dateAdded).format('YYYY-MM-DD'),
				quantity: item.currQuantity,
				note: item.note,
			})
		}
	};
	
	componentWillUnmount = () => {
		// Remove listerner that sets up form to submit on keypress of enter key
		window.removeEventListener("keypress", this.handlePressEnter);
	};

	/*******************************************************************
    RENDER
  *******************************************************************/
  render() {
		const { error } = this.state;
    const { sections = [] } = this.context;
		const { formName } = this.props;

    return (
			<section className={`${formName}__section mainContainer`}>
				<h2>{this.props.formName === 'AddItem' ? 'Add' : 'Edit'} Item</h2>
				<form 
					className={`${formName}__form`}
          onSubmit={(e) => this.handleSubmit(e)}
				>
					<div className={`${formName}__error`} >
            {error && error.message}
            {this.state.missingFields &&
							`Please fill out the following required fields and resubmit: ${this.state.missingFields}`}
					</div>
					

					<div>
						<label htmlFor='name'>
              Name
              <Required />
            </label>
						{' '}
						<input
							type='text'
							name='name'
              id='name'
              defaultValue={this.state.name}
							onChange={e => this.handleInputChange(e)}
							autoFocus
							required
						></input>
					</div>

					<div>
						<label htmlFor='sectionId'>
              Section
              <Required />  
            </label>
						{' '}
						<select
							name='sectionId'
							id='sectionId'
							value={this.state.sectionId}
							onChange={e => this.handleInputChange(e)}
							className={`${formName}__select`}
							required
						>
							<option value=''>Select one</option>
							{sections.map(section => {
								return (
									<option key={section.id} value={section.id}>
									{section.name}
									</option>
								)
							})}
						</select>
					</div>
					
					<div>
						<label htmlFor='dateAdded'>
							Date Added
							<Required />
						</label>
						{' '}
						<input 
							type='date'
							name='dateAdded'
							id='dateAdded'
							value={this.state.dateAdded}
							onChange={e => this.handleInputChange(e)}
							required
						/>
					</div>

					<div>
						<label htmlFor='quantity'>
              Quantity
              <Required />  
            </label>
						{' '}
						<input
							type='number'
							name='quantity'
							id='quantity'
							defaultValue={this.state.quantity}
							onChange={e => this.handleInputChange(e)}
							required
						></input>
					</div>

					<div>
						<label htmlFor='note'>Note</label>
						{' '}
						<textarea
							type='text'
							name='note'
							id='note'
							defaultValue={this.state.note}
							onChange={e => this.handleInputChange(e)}
							className={`${formName}__textarea`}
						></textarea>
					</div>

					<div className={`${formName}__buttonContainer`}>
            <button onClick={e => this.handleClickCancel(e)}>Cancel</button>
            {' '}
            <button type="submit">{`Save`}</button>
					</div>
				</form>
			</section>
		);
  }
}
 