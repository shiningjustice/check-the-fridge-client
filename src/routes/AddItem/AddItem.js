import React from 'react';

import ItemForm from '../../components/ItemForm/ItemForm';

export default function AddItem (props) {
	
	return (
		<ItemForm
			formName='AddItem'
			history={props.history}
		/>
	)
}