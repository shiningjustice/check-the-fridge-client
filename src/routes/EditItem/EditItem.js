import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
// import moment from 'moment';

// import config from '../config';
import ItemForm from '../../components/ItemForm/ItemForm'
import ApiContext from '../../contexts/ApiContext';

// const Required = () => (
//   <span className='ItemForm__Required'>*</span>
// )

class EditItem extends Component {
  static contextType = ApiContext;

  // state = {
	// 	error: null,
	// 	name: '', 
	// 	sectionId: '',
	// 	dateAdded: moment().format('YYYY-MM-DD'),
	// 	quantity: '',
	// 	note: '',
  // };
  
  // handleInputChange = e => {
	// 	const target = e.target;
	// 	const value = target.type === 'checkbox' ? target.checked : target.value;
	// 	const name = target.name;

	// 	console.log(name, value)
	// 	this.setState({
	// 		[name]: value
	// 	});
  // }
  
	// resetState = () => {
	// 	this.setState({
	// 		error: null,
	// 		name: '', 
	// 		sectionId: '',
	// 		dateAdded: moment().format('YYYY-MM-DD'),
	// 		quantity: '',
  //     note: '',
  //     missingFields: [],
	// 	})
	// }

	// setStateForEdit = (name, sectionId, quantity, note,) => {
	// 	this.setState({
	// 		name, sectionId, quantity, note,
	// 	})
	// }

  // handleClickCancel = () => {
	// 	this.resetState();
	// 	this.props.history.push('/dashboard');
  // };

  // addMissingField = field => {
  //   this.setState({
  //     missingFields: [...this.state.missingFields, field]
  //   });
  // }

  // populateForEdit = () => {
	// 	const requiredFields = ['name', 'dateAdded', 'sectionId', 'quantity', 'note'];

  //   for (let i = 0; i > requiredFields.length; i++) {
  //     if (!this.state.requiredFields[i]) {
  //       this.addMissingField(requiredFields[i])
  //     }
  //   }
	// 	// requiredFields.map(field => {
	// 	// 	if (!this.state[field].value) {
	// 	// 		this.addMissingField(field)
	// 	// 	}
	// 	// });
  // };
  
  // handleSubmit = (e) => {
	// 	e.preventDefault();

	// 	this.validateOnSubmit();

	// 	const item = {
	// 		name: this.state.name,
	// 		sectionId: this.state.sectionId,
	// 		dateAdded: moment(this.state.dateAdded).format(),
	// 		quantity: this.state.quantity,
	// 		note: this.state.note
	// 	};

	// 	console.log(item)
	// 	this.setState({ error: null });

	// 	fetch(`${config.API_ENDPOINT}/items`, {
	// 		method: 'PATCH',
	// 		body: JSON.stringify(item),
	// 		headers: {
	// 			"content-type": "application/json"
	// 		},
	// 	})
	// 		.then(res => {
	// 			if (!res.ok) {
	// 				return res.json().then(e => Promise.reject(e));
	// 			}
	// 			return res.json();
  //     })
	// 		.then(data => {
	// 			this.resetState();
	// 			this.context.editItem(data)
	// 			this.props.history.push('/dashboard');
	// 		})
	// 		.catch(error => {
	// 			console.error(error);
	// 			this.setState({ error });
	// 		});
	// };
	



  render() {
    // const { error } = this.state;

    const { items } = this.context;
    // const { sections = [] } = this.context;
    // const { formName } = this.props;

    if (!this.props.match.params.id || items.length === 0) {
      return null
    }

    const itemId = this.props.match.params.id; 
    const item = items.find(item => +item.id === +itemId); 
    // const name = item.name;
    // const sectionId = item.sectionId;
    // const dateAdded = item.dateAdded;
    // const quantity = item.currQuantity;
    // const note = item.note;

  //     return (
  //       <section className={formName}>
  //         <h2>Edit Item</h2>
  //         <form 
  //           className={`EditItem__form`}
  //           onSubmit={(e) => this.handleSubmit(e)}
  //         >
  //           <div className={`EditItem__error`} role='alert'>
  //             {error && <p>{error.message}</p>}
  //             {this.state.missingFields &&
  //               `Please fill out the following required fields and resubmit: ${this.state.missingFields}`}
  //           </div>
            

  //           <div>
  //             <label htmlFor='name'>
  //               Name
  //               <Required />
  //             </label>
  //             {' '}
  //             <input
  //               type='text'
  //               name='name'
  //               id='name'
  //               defaultValue={name && name}
  //               value={this.state.name.value}
  //               onChange={e => this.handleInputChange(e)}
  //               required
  //             ></input>
  //           </div>

  //           <div>
  //             <label htmlFor='sectionId'>
  //               Section
  //               <Required />  
  //             </label>
  //             {' '}
  //             <select
  //               name='sectionId'
  //               id='sectionId'
  //               defaultValue={sectionId && sectionId}
  //               value={this.state.sectionId.value}
  //               onChange={e => this.handleInputChange(e)}
  //               required
  //             >
  //               <option value=''>Select one</option>
  //               {sections.map(section => (
  //                 <option key={section.id} value={section.id}>
  //                   {section.name}
  //                 </option>
  //               ))}
  //             </select>
  //           </div>
            
  //           <div>
  //             <label htmlFor='dateAdded'>
  //               Date Added
  //               <Required />
  //             </label>
  //             {' '}
  //             <input 
  //               type='date'
  //               name='dateAdded'
  //               id='dateAdded'
  //               defaultValue={dateAdded ? moment(dateAdded).format('YYYY-MM-DD') : this.state.dateAdded}
  //               value={this.state.dateAdded.value}
  //               onChange={e => this.handleInputChange(e)}
  //               required
  //             />
  //           </div>

  //           <div>
  //             <label htmlFor='quantity'>
  //               Quantity
  //               <Required />  
  //             </label>
  //             {' '}
  //             <input
  //               type='number'
  //               name='quantity'
  //               id='quantity'
  //               defaultValue={quantity && quantity}
  //               value={this.state.quantity.value}
  //               onChange={e => this.handleInputChange(e)}
  //               required
  //             ></input>
  //           </div>

  //           <div>
  //             <label htmlFor='quantity'>Note</label>
  //             {' '}
  //             <textarea
  //               type='text'
  //               name='note'
  //               id='note'
  //               defaultValue={note && note}
  //               value={this.state.note.value}
  //               onChange={e => this.handleInputChange(e)}
  //             ></textarea>
  //           </div>

  //           <div className={`EditItem__buttonContainer`}>
  //             <button onClick={this.handleClickCancel}>Cancel</button>
  //             {' '}
  //             <button type='submit'>{`Save`}</button>
  //           </div>
  //         </form>
  //       </section>
  //     );
  //   }
  // }
   
    return (
      // <h2>hi</h2>
      <ItemForm
        formName='EditItem'
        history={this.props.history}
        item={item}
      />
    )
  }
}

export default withRouter(EditItem);