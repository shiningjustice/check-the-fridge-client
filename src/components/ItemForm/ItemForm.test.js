import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import ItemForm from './ItemForm';
import EditItem from '../../routes/EditItem/EditItem';
import AddItem from '../../routes/AddItem/AddItem';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <ItemForm />
    </BrowserRouter>
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected for Add Form', () => {
  const tree = renderer
    .create(<ItemForm parent={AddItem} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
it('renders the UI as expected for Edit Form', () => {
  const tree = renderer 
    .create(<ItemForm parent={EditItem} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})