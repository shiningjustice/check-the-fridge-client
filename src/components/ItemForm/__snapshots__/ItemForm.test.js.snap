// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected for Add Form 1`] = `
<section
  className="undefined__section mainContainer"
>
  <h2>
    Edit
     Item
  </h2>
  <form
    className="undefined__form"
    onSubmit={[Function]}
  >
    <div
      className="undefined__error"
      role="alert"
    />
    <div>
      <label
        htmlFor="name"
      >
        Name
        <span
          className="ItemForm__Required"
        >
          *
        </span>
      </label>
       
      <input
        defaultValue=""
        id="name"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div>
      <label
        htmlFor="sectionId"
      >
        Section
        <span
          className="ItemForm__Required"
        >
          *
        </span>
      </label>
       
      <select
        className="undefined__select"
        id="sectionId"
        name="sectionId"
        onChange={[Function]}
        required={true}
        value=""
      >
        <option
          value=""
        >
          Select one
        </option>
      </select>
    </div>
    <div>
      <label
        htmlFor="dateAdded"
      >
        Date Added
        <span
          className="ItemForm__Required"
        >
          *
        </span>
      </label>
       
      <input
        defaultValue="2019-10-24"
        id="dateAdded"
        name="dateAdded"
        onChange={[Function]}
        required={true}
        type="date"
      />
    </div>
    <div>
      <label
        htmlFor="quantity"
      >
        Quantity
        <span
          className="ItemForm__Required"
        >
          *
        </span>
      </label>
       
      <input
        defaultValue=""
        id="quantity"
        name="quantity"
        onChange={[Function]}
        required={true}
        type="number"
      />
    </div>
    <div>
      <label
        htmlFor="note"
      >
        Note
      </label>
       
      <textarea
        className="undefined__textarea"
        defaultValue=""
        id="note"
        name="note"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="undefined__buttonContainer"
    >
      <button
        onClick={[Function]}
      >
        Cancel
      </button>
       
      <button
        type="submit"
      >
        Save
      </button>
    </div>
  </form>
</section>
`;

exports[`renders the UI as expected for Edit Form 1`] = `
<section
  className="undefined__section mainContainer"
>
  <h2>
    Edit
     Item
  </h2>
  <form
    className="undefined__form"
    onSubmit={[Function]}
  >
    <div
      className="undefined__error"
      role="alert"
    />
    <div>
      <label
        htmlFor="name"
      >
        Name
        <span
          className="ItemForm__Required"
        >
          *
        </span>
      </label>
       
      <input
        defaultValue=""
        id="name"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div>
      <label
        htmlFor="sectionId"
      >
        Section
        <span
          className="ItemForm__Required"
        >
          *
        </span>
      </label>
       
      <select
        className="undefined__select"
        id="sectionId"
        name="sectionId"
        onChange={[Function]}
        required={true}
        value=""
      >
        <option
          value=""
        >
          Select one
        </option>
      </select>
    </div>
    <div>
      <label
        htmlFor="dateAdded"
      >
        Date Added
        <span
          className="ItemForm__Required"
        >
          *
        </span>
      </label>
       
      <input
        defaultValue="2019-10-24"
        id="dateAdded"
        name="dateAdded"
        onChange={[Function]}
        required={true}
        type="date"
      />
    </div>
    <div>
      <label
        htmlFor="quantity"
      >
        Quantity
        <span
          className="ItemForm__Required"
        >
          *
        </span>
      </label>
       
      <input
        defaultValue=""
        id="quantity"
        name="quantity"
        onChange={[Function]}
        required={true}
        type="number"
      />
    </div>
    <div>
      <label
        htmlFor="note"
      >
        Note
      </label>
       
      <textarea
        className="undefined__textarea"
        defaultValue=""
        id="note"
        name="note"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="undefined__buttonContainer"
    >
      <button
        onClick={[Function]}
      >
        Cancel
      </button>
       
      <button
        type="submit"
      >
        Save
      </button>
    </div>
  </form>
</section>
`;
